dask_cloudprovider/gcp/instances.py

```diff
@@ -546,24 +546,41 @@ def __init__(
         super().__init__(**kwargs)
 
 
-def authenticate():
-    if os.environ.get("GOOGLE_APPLICATION_CREDENTIALS", False):
-        compute = googleapiclient.discovery.build("compute", "v1")
-    else:
-        import google.auth.credentials  # google-auth
-
-        path = os.path.join(os.path.expanduser("~"), ".config/gcloud/credentials.db")
-        if not os.path.exists(path):
-            raise GCPCredentialsError()
-        conn = sqlite3.connect(path)
-        creds_rows = conn.execute("select * from credentials").fetchall()
-        with tmpfile() as f:
-            with open(f, "w") as f_:
-                # take first row
-                f_.write(creds_rows[0][1])
-            creds, _ = google.auth.load_credentials_from_file(filename=f)
-        compute = googleapiclient.discovery.build("compute", "v1", credentials=creds)
-    return compute
+class GCPCompute:
+    """Wrapper for the ``googleapiclient`` compute object."""
+
+    def __init__(self):
+        self._compute = None
+        self.refresh_client()
+
```

Comment:
What do you think of having `self._compute` explicitly defined in `__init__`  ?

Reply:
Yeah I wasn't sure how to approach this. Should `self.refresh_client()` return the compute object instead of updating it?

Reply:
I think updating it good?  Is this how web token are updated ?

Reply:
I meant should it be `self._compute = self.refresh_client()`?

Reply:
Oh I see, yeah, maybe returning and storing.  This is a small nit so if we should do what you prefer

Reply:
I'll update it. Thanks for the review!